sema4ai/src/sema4ai_code/action_server.py

```diff
@@ -229,3 +236,282 @@ def get_error_message(result_message: str) -> str:
             return ActionResult(
                 success=False, message=get_error_message(command_result.message or "")
             )
+
+    def cloud_login(self, access_credentials: str, hostname: str) -> ActionResult:
+        """
+        Set authentication information to the Control Room for action server.
+
+        Args:
+            access_credentials: Control Room access credentials.
+            hostname: Control Room hostname.
+        """
+        args = [
+            "cloud",
+            "login",
+            "--access-credentials",
+            access_credentials,
+            "--hostname",
+            hostname,
+        ]
+
+        command_result = self._run_action_server_command(args)
+
+        if command_result.success:
+            return ActionResult(success=True, message=None)
+        else:
+            return ActionResult(
+                success=False,
+                message=f"Error login to Control Room.\n{command_result.message or ''}",
+            )
+
+    def verify_login(self) -> ActionResult[ActionServerVerifyLoginResultDict]:
+        """
+        Verify authentication information from the Control Room for action server.
+        """
+        args = ["cloud", "verify-login", "--json"]
+
+        command_result = self._run_action_server_command(args, timeout=ONE_MINUTE_S)
+
+        if command_result.success:
+            try:
+                if command_result.result:
+                    result = json.loads(command_result.result)
+                else:
+                    return ActionResult(
+                        success=False,
+                        message=f"Error empty output.\n{command_result.message}",
+                    )
+            except Exception:
+                return ActionResult(
+                    success=False,
+                    message=f"Error parsing output.\n{command_result.result}",
+                )
+            return ActionResult(success=True, message=None, result=result)
+        else:
+            return ActionResult(
+                success=False,
+                message=f"Error verifying Control Room authentication.\n{command_result.message or ''}",
+            )
+
+    def list_organizations(
+        self, access_credentials: Optional[str], hostname: Optional[str]
+    ) -> ActionResult[ActionServerListOrgsResultDict]:
+        """
+        List available organizations with access credentials used at login.
+        """
+        args = ["cloud", "list-organizations", "--json"]
+
+        if access_credentials and hostname:
+            args.extend(
+                ["--access-credentials", access_credentials, "--hostname", hostname]
+            )
```

Comment:
These should be mandatory here as well, right?
Can we connect to the cloud without these?

Reply:
Nope.